<Type Name="ErrorAddDataBoundRowEventArgs" FullName="Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs">
  <Metadata><Meta Name="ms.openlocfilehash" Value="8ca9565feae3f21930301b7ac9173e83d78767b2" /><Meta Name="ms.sourcegitcommit" Value="7439021cdeee6a9ef5d4640250701344f6cbba15" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="zh-CN" /><Meta Name="ms.lasthandoff" Value="08/13/2019" /><Meta Name="ms.locfileid" Value="69005019" /></Metadata><TypeSignature Language="C#" Value="public interface ErrorAddDataBoundRowEventArgs" />
  <TypeSignature Language="ILAsm" Value=".class public interface auto ansi abstract ErrorAddDataBoundRowEventArgs" />
  <TypeSignature Language="DocId" Value="T:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs" />
  <AssemblyInfo>
    <AssemblyName>Microsoft.Office.Tools.Excel</AssemblyName>
    <AssemblyVersion>10.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Runtime.InteropServices.Guid("eca34a5d-a43c-4be4-a24a-49d49ae4519b")</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="f2614-101">提供有关 <see cref="E:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow" /> 事件的信息。</span><span class="sxs-lookup"><span data-stu-id="f2614-101">Provides information for the <see cref="E:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow" /> event.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f2614-102">如果<xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow>在将新行添加<xref:Microsoft.Office.Tools.Excel.ListObject>到绑定到的<xref:Microsoft.Office.Tools.Excel.ListObject>数据源的过程中发生异常, 则控件将引发该事件。</span><span class="sxs-lookup"><span data-stu-id="f2614-102">The <xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow> event is raised by the <xref:Microsoft.Office.Tools.Excel.ListObject> control if an exception occurs during an attempt to add a new row to the data source bound to the <xref:Microsoft.Office.Tools.Excel.ListObject>.</span></span> <span data-ttu-id="f2614-103">属性可用于获取<xref:Microsoft.Office.Tools.Excel.ListObject>尝试添加的新项。 <xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Item%2A></span><span class="sxs-lookup"><span data-stu-id="f2614-103">The <xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Item%2A> property can be used to get the new item that the <xref:Microsoft.Office.Tools.Excel.ListObject> tried to add.</span></span> <span data-ttu-id="f2614-104">属性提供引发的异常, <xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Retry%2A>属性可用于指示是否应进行其他尝试添加此行。 <xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException%2A></span><span class="sxs-lookup"><span data-stu-id="f2614-104">The <xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException%2A> property provides the exception that was thrown, and the <xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Retry%2A> property can be used to indicate whether another attempt to add this row should be made.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="f2614-105">下面<xref:System.Data.DataTable>的代码示例创建<xref:Microsoft.Office.Tools.Excel.ListObject>和, <xref:System.Data.DataTable>并将绑定<xref:Microsoft.Office.Tools.Excel.ListObject>到。</span><span class="sxs-lookup"><span data-stu-id="f2614-105">The following code example creates a <xref:System.Data.DataTable> and a <xref:Microsoft.Office.Tools.Excel.ListObject>, and binds the <xref:Microsoft.Office.Tools.Excel.ListObject> to the <xref:System.Data.DataTable>.</span></span> <span data-ttu-id="f2614-106">然后, 创建一个<xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow>事件处理程序。</span><span class="sxs-lookup"><span data-stu-id="f2614-106">It then creates an <xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow> event handler.</span></span> <span data-ttu-id="f2614-107">若要测试事件, 请将新行手动添加到<xref:Microsoft.Office.Tools.Excel.ListObject> , 并输入姓氏 "更改" 和名字。</span><span class="sxs-lookup"><span data-stu-id="f2614-107">To test the event, manually add a new row to the <xref:Microsoft.Office.Tools.Excel.ListObject> and enter the last name "Chan" and a first name.</span></span> <span data-ttu-id="f2614-108">事件处理程序将显示一条消息。</span><span class="sxs-lookup"><span data-stu-id="f2614-108">The event handler displays a message.</span></span>  
  
 <span data-ttu-id="f2614-109">此示例适用于文档级自定义项。</span><span class="sxs-lookup"><span data-stu-id="f2614-109">This example is for a document-level customization.</span></span>  
  
 [!code-csharp[Trin_ExcelListObject#20](~/snippets/csharp/VS_Snippets_OfficeSP/Trin_ExcelListObject/CS/Trin_ExcelListObject.cs#20)]
 [!code-vb[Trin_ExcelListObject#20](~/snippets/visualbasic/VS_Snippets_OfficeSP/Trin_ExcelListObject/VB/Sheet1.vb#20)]  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <Member MemberName="InnerException">
      <MemberSignature Language="C#" Value="public Exception InnerException { get; }" />
      <MemberSignature Language="ILAsm" Value=".property instance class System.Exception InnerException" />
      <MemberSignature Language="DocId" Value="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException" />
      <MemberType>Property</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Office.Tools.Excel</AssemblyName>
        <AssemblyVersion>10.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Exception</ReturnType>
      </ReturnValue>
      <Docs>
        <summary><span data-ttu-id="f2614-110">获取导致当前异常的 <see cref="T:System.Exception" /> 实例。</span><span class="sxs-lookup"><span data-stu-id="f2614-110">Gets the <see cref="T:System.Exception" /> instance that caused the current exception.</span></span></summary>
        <value><span data-ttu-id="f2614-111">一个 <see cref="T:System.Exception" /> 的实例，描述导致当前异常的错误。</span><span class="sxs-lookup"><span data-stu-id="f2614-111">An instance of <see cref="T:System.Exception" /> that describes the error that caused the current exception.</span></span></value>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f2614-112">当尝试向数据源中添加新行时,事件将作为异常的直接结果引发。<xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow></span><span class="sxs-lookup"><span data-stu-id="f2614-112">The <xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow> event is raised as a direct result of an exception while attempting to add a new row to the data source.</span></span> <span data-ttu-id="f2614-113"><xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException%2A>使用属性可获取导致此错误的异常。</span><span class="sxs-lookup"><span data-stu-id="f2614-113">Use the <xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException%2A> property to obtain the exception that led to this error.</span></span> <span data-ttu-id="f2614-114"><xref:System.Data.DataTable>例如, 如果绑定<xref:Microsoft.Office.Tools.Excel.ListObject>到控件的不`null`允许使用列, 并且用户未指定此列的值, 则尝试添加行将导致引发一个的<xref:System.Data.NoNullAllowedException> 。</span><span class="sxs-lookup"><span data-stu-id="f2614-114">For example, if the <xref:System.Data.DataTable> that is bound to the <xref:Microsoft.Office.Tools.Excel.ListObject> control does not allow a column to be `null`, and the user does not specify a value for this column, the attempt to add the row causes a <xref:System.Data.NoNullAllowedException> to be raised.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="Item">
      <MemberSignature Language="C#" Value="public object Item { get; }" />
      <MemberSignature Language="ILAsm" Value=".property instance object Item" />
      <MemberSignature Language="DocId" Value="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Item" />
      <MemberType>Property</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Office.Tools.Excel</AssemblyName>
        <AssemblyVersion>10.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Object</ReturnType>
      </ReturnValue>
      <Docs>
        <summary><span data-ttu-id="f2614-115">获取 <see cref="T:Microsoft.Office.Tools.Excel.ListObject" /> 尝试添加到数据源的新项。</span><span class="sxs-lookup"><span data-stu-id="f2614-115">Gets the new item that the <see cref="T:Microsoft.Office.Tools.Excel.ListObject" /> attempted to add to the data source.</span></span></summary>
        <value><span data-ttu-id="f2614-116">要添加的项。</span><span class="sxs-lookup"><span data-stu-id="f2614-116">The item to be added.</span></span></value>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f2614-117">当绑定<xref:System.Data.DataRowView>到时, 项将<xref:System.Data.DataRow>是而不是。 <xref:System.Data.DataTable> <xref:Microsoft.Office.Tools.Excel.ListObject></span><span class="sxs-lookup"><span data-stu-id="f2614-117">When the <xref:Microsoft.Office.Tools.Excel.ListObject> is bound to a <xref:System.Data.DataTable>, the item would be a <xref:System.Data.DataRowView> rather than a <xref:System.Data.DataRow>.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="Retry">
      <MemberSignature Language="C#" Value="public bool Retry { get; set; }" />
      <MemberSignature Language="ILAsm" Value=".property instance bool Retry" />
      <MemberSignature Language="DocId" Value="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Retry" />
      <MemberType>Property</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Office.Tools.Excel</AssemblyName>
        <AssemblyVersion>10.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Boolean</ReturnType>
      </ReturnValue>
      <Docs>
        <summary><span data-ttu-id="f2614-118">获取或设置一个值，该值指示 <see cref="T:Microsoft.Office.Tools.Excel.ListObject" /> 是否应该再次尝试添加此项。</span><span class="sxs-lookup"><span data-stu-id="f2614-118">Gets or sets a value indicating whether the <see cref="T:Microsoft.Office.Tools.Excel.ListObject" /> should attempt to add the item again.</span></span></summary>
        <value><span data-ttu-id="f2614-119"><see langword="true" />如果应进行其他尝试, 则为;否则为<see langword="false" />。</span><span class="sxs-lookup"><span data-stu-id="f2614-119"><see langword="true" /> if another attempt should be made; otherwise, <see langword="false" />.</span></span> <span data-ttu-id="f2614-120">默认值为 <see langword="false" />。</span><span class="sxs-lookup"><span data-stu-id="f2614-120">The default value is <see langword="false" />.</span></span></value>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f2614-121">如果尝试添加新行时<xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow> 遇到错误,则会引发事件。<xref:Microsoft.Office.Tools.Excel.ListObject></span><span class="sxs-lookup"><span data-stu-id="f2614-121">If the <xref:Microsoft.Office.Tools.Excel.ListObject> encounters an error when attempting to add the new row, the <xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow> event is raised.</span></span> <span data-ttu-id="f2614-122">开发人员可以处理此事件并尝试修复发生的任何错误。</span><span class="sxs-lookup"><span data-stu-id="f2614-122">The developer can handle this event and attempt to fix any errors that occur.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>
